### PR TITLE
Fix incorrect results returned by `min`, `max` and `avg`

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
Fix incorrect results returned by `min`, `max` and `avg` when processing null/missing values only.

Signed-off-by: Yury-Fridlyand <yuryf@bitquilltech.com>

### Description
**Before**
```
opensearchsql> select min(int0) from calcs where int0 IS NULL;
fetched rows / total rows = 1/1
+-------------+
| min(int0)   |
|-------------|
| 2147483647  |
+-------------+
```
**After**
```
opensearchsql> select min(int0) from calcs where int0 IS NULL;
fetched rows / total rows = 1/1
+-------------+
| min(int0)   |
|-------------|
| null        |
+-------------+
```
### Fixed:
* `min`
* `max`
* `avg`

### Not fixed:
* `sum`
```
opensearchsql> select sum(int0) from calcs where int0 IS NULL;
fetched rows / total rows = 1/1
+-------------+
| sum(int0)   |
|-------------|
| 0           |
+-------------+
```
```
mysql> select sum(int0) from calcs where int0 IS NULL;

+-----------+
| sum(int0) |
+-----------+
|      NULL |
+-----------+
1 row in set (0.00 sec)
```
This requires much more changes.

### Not affected
* `var_samp`
* `var_pop`
* `stddev_samp`
* `stddev_pop`
 
### Issues Resolved
https://github.com/opensearch-project/sql/issues/817
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: integ-test/src/test/resources/indexDefinitions/null_missing_index_mapping.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            },
+            "str" : {
+                "type" : "text"
+            },
+            "date" : {
+                "type" : "date",
+                "format": "yyyy-MM-dd"
+            },
+            "time" : {
+                "type" : "date",
```
Comment (acarbonetto) [inline integ-test/src/test/resources/indexDefinitions/null_missing_index_mapping.json:24]: 😆

File: integ-test/src/test/resources/null_missing.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{"index":{}}
```
Comment (acarbonetto) [inline integ-test/src/test/resources/null_missing.json:1]: you don't want to give these index numbers?

File: integ-test/src/test/resources/null_missing.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+{"key" : "values", "int" : 42, "dbl" : 3.1415, "bool" : true, "str" : "pewpew", "date" : "1984-05-22", "time" : "22:13:37", "datetime" : "1984-05-22 22:13:37", "timestamp" : "2000-01-02 03:04:05" }
+{"index":{}}
+{"key" : "missing"}
```
Comment (acarbonetto) [inline integ-test/src/test/resources/null_missing.json:4]: is this not confusing?

File: integ-test/src/test/resources/null_missing.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+{"index":{}}
+{"key" : "missing"}
+{"index":{}}
+{"key" : "missing"}
+{"index":{}}
+{"key" : "missing"}
+{"index":{}}
+{"key" : "null", "int" : null, "dbl" : null, "bool" : null, "str" : null, "date" : null, "time" : null, "datetime" : null, "timestamp" : null}
```
Comment (acarbonetto) [inline integ-test/src/test/resources/null_missing.json:12]: is `"key": "null"` also not confusing?

**Comment by acarbonetto** [comment]
What the MySQL result for sum(null)?  Is it supposed to be null or zero?

**Review by acarbonetto — APPROVED** [review]

File: integ-test/src/test/resources/null_missing.json (hunk 2 above)
Comment (Yury-Fridlyand) [reply integ-test/src/test/resources/null_missing.json:1]: They assigned automatically if not given.

File: integ-test/src/test/resources/null_missing.json (hunk 3 above)
Comment (Yury-Fridlyand) [reply integ-test/src/test/resources/null_missing.json:4]: `key` is not a PK. Just a column.

**Comment by Yury-Fridlyand** [comment]
> What the MySQL result for sum(null)? Is it supposed to be null or zero?

```
mysql> select sum(num0) from calcs where num0 is null;
+-----------+
| sum(num0) |
+-----------+
|      NULL |
+-----------+
1 row in set (0.00 sec)
```

**Review by GumpacG — APPROVED** [review]

**Review by forestmvey — APPROVED** [review]

**Review by MitchellGale-BitQuill — APPROVED** [review]
LGTM

**Pull request merged**